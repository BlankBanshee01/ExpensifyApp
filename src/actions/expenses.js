import databse from "../firebase/firebase";

// ADD_EXPENSE
export const addExpense = (expense) => {
  return {
    type: "ADD_EXPENSE",
    expense,
  };
};

export const startAddExpense = (expenseData = {}) => {
  return (dispatch) => {
    const {
      description = "",
      note = "",
      amount = 0,
      createdAt = 0,
    } = expenseData;

    const expense = { description, note, amount, createdAt };
    return databse
      .ref("expenses")
      .push(expense)
      .then((ref) => {
        dispatch(addExpense({ id: ref.key, ...expense }));
      });
  };
};

// REMOVE_EXPENSE
export const removeExpense = ({ id } = {}) => ({
  type: "REMOVE",
  id,
});

// EDIT_EXPENSE
export const editExpense = (id, changes) => ({
  type: "EDIT_EXPENSE",
  id,
  changes,
});

//SET_EXPENSES

export const setExpenses = (expenses) => ({
  type: "SET_EXPENSES",
  expenses,
});

// START_SET_EXPENSE

export const startSetExpenses = () => {
  return (dispatch) => {
    const expenses = [];
    return databse
      .ref("expenses")
      .once("value")
      .then((snapshot) => {
        snapshot.forEach((childSnapshot) => {
          expenses.push({ id: childSnapshot.key, ...childSnapshot.val() });
        });
        dispatch(setExpenses(expenses));
      });
  };
};
