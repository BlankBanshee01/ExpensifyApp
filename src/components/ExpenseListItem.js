import React from "react";
import { Link } from "react-router-dom";

const expenseListItem = ({ description, amount, createdAt, id }) => {
  return (
    <div>
      <Link to={`/edit/${id}`}>
        <h3>{description}</h3>
      </Link>
      <p>Amount: {amount}</p>
      <p>Date: {createdAt}</p>
    </div>
  );
};

export default expenseListItem;
