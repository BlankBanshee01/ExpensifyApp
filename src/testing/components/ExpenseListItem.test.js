import React from "react";
import { shallow } from "enzyme";
import ExpenseListItem from "../../components/ExpenseListItem";
import expenses from "../fixtures/expenses";

test("should render expense list item", () => {
  const render = shallow(<ExpenseListItem {...expenses[0]} />);
  expect(render).toMatchSnapshot();
});
