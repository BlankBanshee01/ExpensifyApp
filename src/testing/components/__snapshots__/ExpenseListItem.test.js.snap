// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render expense list item 1`] = `
<div>
  <Link
    to="/edit/1"
  >
    <h3>
      Gum
    </h3>
  </Link>
  <p>
    Amount: 
    195
  </p>
  <p>
    Date: 
    0
  </p>
  <p>
    Note: 
  </p>
</div>
`;
